core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -1322,4 +1336,38 @@ object Chunk {
     private val CharClass       = classTag[Char]
     private val CharClassBox    = classTag[java.lang.Character]
   }
+
+  final case class RefChunkArr[A <: AnyRef](array: Array[A]) extends Arr[A]
```

Comment:
Last naming request - can we call the chunks `Refs, Bytes, Ints` etc. for brevity?